core/src/file/cfile.cc

```diff
@@ -173,15 +177,23 @@ long cfile::tell() {
  *  @return Number of bytes written.
  */
 long cfile::write(void const* buffer, long size) {
-  if (!_stream)
-    _open();
+  assert(_stream);
+//  if (!_stream)
+//    _open();
   size_t retval(fwrite(buffer, 1, size, _stream));
   if (ferror(_stream)) {
     char const* msg(strerror(errno));
-    throw(exceptions::msg()
-          << "cannot write " << size << " bytes to file: " << msg);
+    throw exceptions::msg()
+          << "cannot write " << size << " bytes to file: " << msg;
   }
-  return (retval);
+  return retval;
+}
+
+/**
+ *  Flush the opened stream.
+ */
+void cfile::flush() {
+  fflush(_stream);
```

Comment:
you did not check the return value here. But I do not know if there is something to do if fflush fails.